CMakeLists.txt

```diff
@@ -532,8 +532,8 @@ endif()
 # declared earlier, so we create binaryen_wasm/js.js, which must then be copied.
 if(EMSCRIPTEN)
   # binaryen.js WebAssembly variant
-  add_executable(binaryen_wasm ${binaryen_SOURCES})
-  target_link_libraries(binaryen_wasm PRIVATE binaryen)
+  add_executable(binaryen_wasm ${binaryen_HEADERS})
+  target_link_libraries(binaryen_wasm PRIVATE "$<LINK_LIBRARY:WHOLE_ARCHIVE,binaryen>")
```

Comment:
Please add a comment about why whole-archive is needed.

Does it work cross-platform?